data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/CsaProfileRemedialActionsCreator.java

```diff
@@ -247,27 +250,45 @@ private void addOnContingencyStateUsageRules(PropertyBag parentRemedialActionPro
         addOnConstraintUsageRules(crac.getInstant(InstantKind.CURATIVE), remedialActionAdder, remedialActionId, alterations);
         boolean isLinkedToAssessedElements = elementaryActionsHelper.remedialActionIsLinkedToAssessedElements(remedialActionId);
         if (!isLinkedToAssessedElements) {
-            String instantId = isAuto ? crac.getInstant(InstantKind.AUTO).getId() : crac.getInstant(InstantKind.CURATIVE).getId();
+            Instant instant = defineInstant(isSchemeRemedialAction, parentRemedialActionPropertyBag, remedialActionId, durationLimit);
 
             validContingencies.forEach(openRaoContingencyId -> remedialActionAdder.newOnContingencyStateUsageRule()
-                .withInstant(instantId)
+                .withInstant(instant.getId())
                 .withContingency(openRaoContingencyId.getLeft())
-                .withUsageMethod(getUsageMethod(openRaoContingencyId.getRight(), isAuto)).add());
+                .withUsageMethod(getUsageMethod(openRaoContingencyId.getRight(), isSchemeRemedialAction, instant)).add());
 
             alterations.addAll(ignoredContingenciesMessages);
         }
     }
 
-    private UsageMethod getUsageMethod(ElementCombinationConstraintKind elementCombinationConstraintKind, boolean isAuto) {
-        if (isAuto) {
-            return UsageMethod.FORCED;
+    private Instant defineInstant(boolean isSchemeRemedialAction, PropertyBag parentRemedialActionPropertyBag, String remedialActionId, int durationLimit) {
+        if (isSchemeRemedialAction) {
+            return crac.getInstant(InstantKind.AUTO);
+        }
+        String timeToImplement = parentRemedialActionPropertyBag.get(TIME_TO_IMPLEMENT);
+        if (timeToImplement == null) {
+            return crac.getInstant(InstantKind.CURATIVE);
         }
-        return ElementCombinationConstraintKind.INCLUDED.equals(elementCombinationConstraintKind) ? UsageMethod.FORCED : UsageMethod.AVAILABLE;
+        int durationInSeconds = 0;
+        try {
+            durationInSeconds = CsaProfileCracUtils.convertDurationToSeconds(timeToImplement);
+        } catch (RuntimeException e) {
+            throw new OpenRaoImportException(ImportStatus.INCONSISTENCY_IN_DATA, "Remedial action " + remedialActionId + " will not be imported because of an irregular timeToImplement pattern");
+        }
+        if (durationInSeconds <= durationLimit) {
+            return crac.getInstant(InstantKind.AUTO);
+        } else {
+            return crac.getInstant(InstantKind.CURATIVE);
+        }
+    }
+
+    private UsageMethod getUsageMethod(ElementCombinationConstraintKind elementCombinationConstraintKind, boolean isSchemeRemedialAction, Instant instant) {
+        return isSchemeRemedialAction || ElementCombinationConstraintKind.INCLUDED.equals(elementCombinationConstraintKind) || instant.isAuto() ? UsageMethod.FORCED : UsageMethod.AVAILABLE;
```

Comment:
This should be the case for auto PST range actions only:
- at the auto instant, PST SPS must be FORCED
- at the auto instant, topo SPS can be either FORCED or AVAILABLE

Reply:
ok, I've updated it to differentiate PST SPS and topo